### DISCUSSION TITLE
Akkustatusanzeige

### DISCUSSION THREAD

**Discussion by KSMBond-CSS** [OpenCCU/OpenCCU#1911] [🇩🇪 Ideen]
Guten Morgen,

Des öfteren ist bei mir der Strom ausgefallen, und jedesmal wird, nach dem hochfahren der CCU3 (RPi4) eine Fehlermeldung angezeigt (ist auch richtig so).
Ich möchte einen Akku nachzurüsten, um einen kleinen Puffer zu haben.
Ist es möglich den Ladezustand auf der Startseite der RaspberryMatic anzeigen zu lassen (da steht sowieso kaum etwas drin)?
Es gibt entsprechende Module zum Einstecken auf die Platine schon zu kaufen (z.B. mit 18650 Lithium-Batterie) Theoretisch müßten sie zwischen dem RPi4 und dem RPI-RF-MOD (4.4.22) gesteckt werden können.

Klaus

**Comment by Baxxy13** [comment]
So einfach ist das nicht. 
Zumindest nicht wenn irgendeine Kommunikation zwischen "Mini-USV" und Raspi stattfinden soll.
Da müsste zuerst geprüft werden ob das Teil die GPIO-Funkmodulkommunikation nicht stört und außerdem müsste Jens einen Treiber / daemon implementieren der dann die Kommunikation übernimmt.
Das ist ne Menge Aufwand und lohnt sich nur bei "Mainstream Geräten" die dann auch verfügbar, funktional und gut dokumentiert sind.
Außerdem kämst du um eine Hardware-Leihe an Jens nicht drum herum, damit er das sauber umsetzten könnte.
Sowas haben wir damals für die Argon1 - Gehäuse / Argon-FAN Lüftersteuerung gemacht. Das läuft heute wunderbar.

"Besser" wäre gleich eine richtige USV die per USB angeschlossen wird. Die Dienste dafür sind in RaspberryMatic integriert und müssen nur konfiguriert werden.

Wenn du also was "kleines" mit USB und HID-Profil findest ließe sich das schnell und einfach nutzen.

Der Weg zu neuer Hardwareunterstützung in RM hingegen ist lang und manchmal holprig. Das heißt aber nicht das es unmöglich ist. ;-)

Aktuell von RM unterstützt:
NUT
Pico UPS
PiUSV+
StromPi2
S.USV

Ob davon wirklich alles mit jedem Funkmodul geht kann ich nicht sagen.

**Comment by akula3** [comment]
Hier meine Lösung für S.USV
![grafik](https://user-images.githubusercontent.com/58461189/186463992-10806068-1f33-439d-a9f1-97ae1a92a31d.png)
Allerdings man muss nach jedem RM Update einiges in/an Files manuell anpassen

**Reply by jens-maus** [reply]
> Allerdings man muss nach jedem RM Update einiges in/an Files manuell anpassen

Könntest du ja mal bei Gelegenheit etwas näher darlegen, dann könnte ich schauen ob man da was dran drehen kann damit man nicht nach jedem Update irgendwelche S.USV Konfigurationen anpassen muss.

**Comment by Baxxy13** [comment]
Sieht ja nicht schlecht aus. 
Und das Funkmodul ist auf der "S.USV tinker mobile" oben drauf?

Aber die Preise sind ja heftig... da bleibe ich lieber bei so nem dicken USV-Klotz. ;-)
![RM_BRICS_USV_Übersicht](https://user-images.githubusercontent.com/54898249/186475229-84fec18f-6acc-405b-bfa1-e5624dfb69a4.JPG)

**Comment by akula3** [comment]
> Und das Funkmodul ist auf der "S.USV tinker mobile" oben drauf?

Ja, ganz genau.

> Könntest du ja mal bei Gelegenheit etwas näher darlegen, dann könnte ich schauen ob man da was dran drehen kann damit man nicht nach jedem Update irgendwelche S.USV Konfigurationen anpassen muss.
1. In der Config Datei muss der GPIO auf 166 für TinkerboardS umgeändert werden.
2. Die Daemon Dateien (twiboot, susv, susvd) in der RM Firmware sind auch schon etwas älter, da gibt es bei Olmatic was neueres.

Es wäre fantastisch, wenn Du beides in der Firmware irgendwie berücksichtigen würdest.

Vielen Dank an der Stelle nochmals an Dich und die Mitwirkenden für die TOLLE Arbeit!

**Reply by jens-maus** [reply]
> > Könntest du ja mal bei Gelegenheit etwas näher darlegen, dann könnte ich schauen ob man da was dran drehen kann damit man nicht nach jedem Update irgendwelche S.USV Konfigurationen anpassen muss.
> 
> 1. In der Config Datei muss der GPIO auf 166 für TinkerboardS umgeändert werden.

Meinst du diese Zeilen hier?

https://github.com/jens-maus/RaspberryMatic/blob/dfb250784f99090abdd89544ddbfda775f24977b/buildroot-external/package/susvd/rootfs-overlay/opt/susvd/susv.cfg#L21-L22

In der Tat könnte ich theoretisch da auch was einbauen das er automatisch erkennt das das ein Tinkerboard oder RaspberryPi ist und dann jeweils den richtigen GPIO angeben.

> 2. Die Daemon Dateien (twiboot, susv, susvd) in der RM Firmware sind auch schon etwas älter, da gibt es bei Olmatic was neueres.

Bist du sicher? Also hier (https://s-usv.de/downloads/) sehe ich nur die 2.40 Version für RaspberryPi und Tinkerboard und die sollte schon länger dabei sein. Auch die S.USV Firmware die ich in RM mit ausliefere sollte eigentlich genau die sein die auf dieser download-seite zu sehen ist?!?!

**Reply by akula3** [reply]
zu 1. - ja
zu 2. - das müsste ich beim nächsten Update probieren. Auf jeden Fall, als ich letztes Jahr den S.USV eingerichtet hatte, die Dateien aus RM Firmware haben nicht funktioniert, ich musste zumindest die susv austauschen. Ich hatte sicherheitshalber die Dateien aus RM mit denen von Olmatic verglichen, und die alle waren unterschiedlich.

**Reply by akula3** [reply]
@jens-maus, ich habe mit der neusten 3.65.8.20220831 ausprobiert, ohne die `susv` auszutauschen. Funktioniert nicht. Musste die originale von Olmatic nehmen.
Liegt wahrscheinlich daran, dass die RM beinhaltet die Dateien für Raspberry's S.USV, und die für Tinkerboard sind anders. Trotz dass sie die gleiche Versionsnummer haben.

**Reply by jens-maus** [reply]
@akula3 Wie schon zuvor gesagt: Bitte das genaue Archiv benennen (oder download link zeigen) das du da verwendest um die s.usv binaries zu ersetzen usw. Und natürlich verwende ich für den Tinkerboard Build andere binaries als für den RaspberryPi Build von RaspberryMatic.

**Reply by akula3** [reply]
@jens-maus hier der link:
http://downloads.s-usv.de/files/firmware/susv_fw_22_tinker_mobile.tar.gz

**Reply by akula3** [reply]
oh, sorry! Das war der link zu der Firmware, hier der zum Daemon:
http://downloads.s-usv.de/files/software/susvd-en-2.40-systemd-tinker.tar.gz

**Reply by jens-maus** [reply]
Ok, danke für die Infos. Hab dazu nun mal ein entsprechendes Issue dafür aufgemacht damit das nicht in vergessenheit gerät: https://github.com/jens-maus/RaspberryMatic/issues/1920

**Comment by KSMBond-CSS** [comment]
![H1ab1379986344e6eb1cbcb92e5c46ccfN](https://user-images.githubusercontent.com/57062176/186486839-ea1b849c-5f7b-4bbc-8198-9a05bacfc8af.jpg)
Es ist keine Externe "USV", sondern eine interne mit zwei Akkus mit den 18650-Akkus (siehe Bild). Ich kann mal nachfragen ob es entsprechende Schnittstellen zum Auslesen gibt. 
So wie es aber scheint, ist keine weitere Software nötig, einfach stecken und dann könnte es eine Stütze sein. Das Heißt für mich "Pin" auslesen.

**Reply by Baxxy13** [reply]
Dürfte dann sowas -> [Waveshare UPS HAT](https://www.waveshare.com/wiki/UPS_HAT) sein.

**Reply by jens-maus** [reply]
Wenn ich den dazugehörigen Schaltplan richtig verstehe (https://www.waveshare.com/w/upload/6/62/UPS_HAT_Schematic.pdf), dann sollte dieses "UPS HAT" prinzipiell mit einem `HM-MOD-RPI-PCB` bzw. `RPI-RF-MOD` funktionieren weil es die GPIO pins die die Funkmodule benötigen einfach durchschleift und selbst nicht verwendet. Allerdings wird die dort veröffentlichte Python Software natürlich unter RaspberryMatic nicht out-of-the-box funktionieren und ob dieser massive Aufbau inkl. `RPI-RF-MOD` on-top dann ggf. irgendwelche zusätzliche Funkstörungen verursacht müsste man erst einmal testen. Auch ob die USV Leistung überhaupt ausreicht um stabil den Pi inkl. RPI-RF-MOD weiter zu versorgen müsste man erst einmal testen.

**Reply by KSMBond-CSS** [reply]
Da ich es gekauft habe, kann ich es soweit Testen (zumindest den Stromausfall).
Bei der Funkentstörung hilft ein Metallgehäuse nach Außerhalb. Ob die Ladeeinheit Intern des RPI stört kann ich nicht testen :( aber merken, da keine Signale mehr eingehen, bzw. richtig verarbeitet werden.
Solch ein Model gibt es auch für bis zu 10 Akkus, aber leider sind keine Gehäuse verfügbar, die auch das Funkmodul beherbergen könnten. Wenn müßte man eines herstellen lassen. Noch denke ich aber sind 10 Akkus zu viel.

Wäre es den möglich ein "Autoshutdown" einzupflegen, wenn Strom ausfällt und der Akku die Leistung übernimmt? Somit wäre ein Ordnungsgemäßes runter/rauffahren ohne Fehlermeldungen möglich. Dies würe die Micro-SD auch schonen (denke ich).

**Reply by jens-maus** [reply]
> Wäre es den möglich ein "Autoshutdown" einzupflegen, wenn Strom ausfällt und der Akku die Leistung übernimmt? Somit wäre ein Ordnungsgemäßes runter/rauffahren ohne Fehlermeldungen möglich. Dies würe die Micro-SD auch schonen (denke ich).

Ob das möglich ist müsste man sich im Detail mit diesem UPS HAT dann anschauen. Kann ich aber nicht, da ich ja dieses UPS HAT hier nicht selbst habe.

**Reply by KSMBond-CSS** [reply]
mmhhh ich wäre bereit Das Modul leiweise weiter zu leiten, wenn ich es bekommen habe, um solche Test machen zu können :))

**Reply by KSMBond-CSS** [reply]
Damit kann nur die Software vom Matic-CCU3 verbessert werden :)

**Comment by KSMBond-CSS** [comment]
Der UPS HAT ist nun eingebaut. Leider muß ich nun das Gehäuse auf lassen, da es zu klein ist :( Demnächst hole ich mir ein anderes :)
Leider scheint es nicht das zu machen, für was es gedacht ist :( Es soll IP4 for einem Stromausfall schützen. Aber der erste Test war nicht erfolgreich :( Die CCU3 ist neu gestartet. 
Die CCU3 macht ihre Arbeit :))
![IMG_0594](https://user-images.githubusercontent.com/57062176/189716554-e796faa8-d113-456b-8be2-2ff8c4c74d7a.JPEG)

**Reply by KSMBond-CSS** [reply]
Könnt ihr mit dem Modul mehr anfangen? Das wäre der nächste Test den ich kaufen würde. Laut Beschreibung dann unterbrechungsfrei :)
https://wiki.geekworm.com/X728?spm=a2g0o.detail.1000023.20.4e3d593eYkNZWW

**Reply by Baxxy13** [reply]
Hast du das Teil mal mit einem Raspberrypi-OS getestet? 
Und das Python-Script dazu installiert um mal zu gucken was geht?

>Aber der erste Test war nicht erfolgreich :( Die CCU3 ist neu gestartet.

Was heißt das? Stecker gezogen und dann Pi4B tot?
Oder Batterien leer und dann tot?
Und wieso CCU3? Zu erkennen ist ein Pi4B.
Und wieso 2 Stromanschlüsse? Sollte nicht der Anschluss am UPS HAT reichen? 

Ein geordnetes herunterfahren kann es nicht geben ohne das Teil über i2c auszulesen.

**Reply by KSMBond-CSS** [reply]
1. warum regst du dich so auf?
2. Ich habe die stecker gezogen, ja.
3. Die Batterien sollten aufgeladen sein, zumindest hatte ich es gemacht.
4. Es ist auch ein Pi4 in Funktion einer CCU3, drum auch das Modul wo die blaue LED leuchtet.
5. Der eine für den PI4 und den anderen für die UPS.
6. Ich kann nicht programmieren. Ich bin von den Programmierer dieser Seite sozusagen abhängig, das sie meine Wünsche verstehen und eventuell umsetzen können :)
7. Es ist die Firmware für Homatic-Geräte (CCU3) aufgespielt.

Nachtrag: Jetzt ging es mit der UPS. Habe beide Stecker entfernt und die UPS hat strom geliefert :) Habe auch mal getestet ob ein Netzteil reicht. Macht es und nun ist nur eines angeschlossen :))

Jetzt muß noch das mit dem Ordentlichen runterfahren funktionieren ..... Da aber auch Router, Modems und Switche stromlos sind, bei einem Ausfall kommt man nicht auf die CCU rauf. 
Ich habe eine 8-Fach Fernbedienung (HmIP-RC8) könnte ich damit eine Taste zum runterfahren der CCU3 programmieren?

**Reply by Baxxy13** [reply]
Das war kein Aufregen...
Mit deinen Informationen konnte man nicht viel anfangen und der etwas rauere Ton meinerseits hat nun dafür gesorgt das verwertbare Informationen gekommen sind. Man weiß jetzt das das Teil grundsätzlich funktioniert. Sehr gut.

Es bleibt aber dabei:
RaspberryMatic hat aktuell keinerlei Zugriff auf die Daten (speziell Batterieladestand) des USV-HAT.
Das Script dazu läuft unter Python was es auf RM nicht gibt. [Test und Scriptbeispiel](https://notenoughtech.com/raspberry-pi/waveshare-ups-hat/) 
Daher gibt es m.E. aktuell keine Möglichkeit das System bei niedrigem Batteriestand gezielt automatisch herunterzufahren.
"Ändern" kann daran nur @jens-maus was, wenn's gut läuft.

"Manuell" per RC8 die Zentrale herunterzufahren ist einfach. 
Simples Programm:
WENN
RC8 - Taste:X - Tastendruck
DANN
Script:
`system.Exec("poweroff");`

**Reply by KSMBond-CSS** [reply]
Ja, ich habe da etwas in einem Homic-Forum (https://homematic-forum.de/forum/viewtopic.php?t=46313) gefunden :))
system.Save();
string stdout;string stderr;
string cmd = "/sbin/poweroff"; 
Damit schaltet er per Fernbedienung sich ab :))

Als Zwischenlösung kann man es anwenden :) Wenn Jens-Maus eventuell das Modul zum testen braucht muß er es sagen :) Schicke ich zum Test und Leihweise ihm :)

**Reply by jens-maus** [reply]
Wenn Jens-Maus eventuell das Modul zum testen braucht muß er es sagen :) Schicke ich zum Test und Leihweise ihm :)

Tut mir leid, leihweise macht zuviel aufwand mit hin/herschicken und wenn man den Preis dieses HAT siehst lohnt sich das hin/herschicken erst recht nicht..

**Reply by KSMBond-CSS** [reply]
Es war auch nur ein Vorschlag meiner seits :)
